src/main/java/com/adriansoftware/BankValueHistoryTracker.java

```diff
@@ -234,7 +234,7 @@ public void addEntry(boolean force, Consumer<String> callback)
 		{
 			int currentBankTab = client.getVar(Varbits.CURRENT_BANK_TAB);
 			LocalDateTime lastEntry = getLastDataEntry(client.getUsername(), currentBankTab);
-			LocalDateTime nextUpdateTime = LocalDateTime.now().plusHours(config.getDefaultDatasetEntry());
+			LocalDateTime nextUpdateTime = lastEntry.plusHours(config.getDefaultDatasetEntry());
```

Comment:
I think `getLastDataEntry` returns null if there are no dataset entries. Could we update this to prevent a NPE?

Reply:
I can add a check for null when settings `nextUpdateTime` but it'll need to be set to something default otherwise. I'm not familiar with `LocalDateTime`s, is there a default value I can set it to such that `.isAfter()` will always be true (like setting to epoch time)?

Reply:
We could probably just make 'nextUpdateTime` null by default. If it's null, we can add an additional entry.

Reply:
I think this might be simplest.
```
LocalDateTime lastEntry = getLastDataEntry(client.getUsername(), currentBankTab);
LocalDateTime nextUpdateTime = LocalDateTime.of(0, 0, 0, 0, 0);
if (lastEntry != null)
{
	nextUpdateTime = lastEntry.plusHours(config.getDefaultDatasetEntry());
}

Reply:
Looks good to me.

Reply:
I accidentally merged this thinking your code was in the commit. I'll add it.

Added in commit [5290e1d1cd5ffa4c1bc3661a9d511d59ebed6eab](https://github.com/AdrianLeeElder/runelite-bank-history-plugin/commit/5290e1d1cd5ffa4c1bc3661a9d511d59ebed6eab)

Reply:
Haha oops. I wanted to wait for your thoughts before pushing it